mlos_bench/mlos_bench/tunables/tunable.py

```diff
@@ -618,41 +606,45 @@ def quantized_values(self) -> Optional[Union[Iterable[int], Iterable[float]]]:
         """
         num_range = self.range
         if self.type == "float":
-            if not self._quantization:
+            if not self.cardinality:
                 return None
             # Be sure to return python types instead of numpy types.
-            cardinality = self.cardinality
-            assert isinstance(cardinality, int)
             return (
                 float(x)
                 for x in np.linspace(
                     start=num_range[0],
                     stop=num_range[1],
-                    num=cardinality,
+                    num=self.cardinality,
                     endpoint=True,
                 )
             )
         assert self.type == "int", f"Unhandled tunable type: {self}"
-        return range(int(num_range[0]), int(num_range[1]) + 1, int(self._quantization or 1))
+        return range(
+            int(num_range[0]),
+            int(num_range[1]) + 1,
+            int(self.span / self.quantization) if self.quantization else 1,
+        )
 
     @property
-    def cardinality(self) -> Union[int, float]:
+    def cardinality(self) -> Optional[int]:
         """
-        Gets the cardinality of elements in this tunable, or else infinity.
+        Gets the cardinality of elements in this tunable, or else None. (i.e., when the
+        tunable is continuous float and not quantized).
 
         If the tunable has quantization set, this
 
         Returns
         -------
-        cardinality : int, float
-            Either the number of points in the tunable or else infinity.
+        cardinality : int
+            Either the number of points in the tunable or else None.
         """
         if self.is_categorical:
             return len(self.categories)
-        if not self.quantization and self.type == "float":
-            return np.inf
-        q_factor = self.quantization or 1
-        return int(self.span / q_factor) + 1
+        if self.quantization:
+            return self.quantization + 1
```

Comment:
Is this accurate?

`quantization` is now equivalent to `number of bins` right?

Isn't that the same as `cardinality`?

Reply:
I think you're right. I was trying to shoehorn the new quantization property semantics into the existing code and unit tests and did not realize how ugly that one-off quantization vs. cardinality now looks like. I'll fix it now.

Reply:
fixed!